applications/FluidDynamicsApplication/custom_strategies/strategies/fs_strategy.h

```diff
@@ -228,164 +154,166 @@ class FSStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinearSolver
+        for (const auto& r_condition : r_model_part.Conditions()) {
+            ierr = r_condition.Check(r_current_process_info);
+            if (ierr != 0) {
+                break;
+            }
         }
 
         return ierr;
 
         KRATOS_CATCH("");
     }
 
-    double Solve() override
+    void InitializeSolutionStep() override
     {
         // Initialize BDF2 coefficients
-        ModelPart& rModelPart = BaseType::GetModelPart();
-        this->SetTimeCoefficients(rModelPart.GetProcessInfo());
-
-        double NormDp = 0.0;
-
-        if (mPredictorCorrector)
-        {
-            bool Converged = false;
+        SetTimeCoefficients();
+    }
 
+    bool SolveSolutionStep() override
+    {
+        double norm_dp = 0.0;
+        bool converged = false;
+        if (mPredictorCorrector) {
+            const unsigned int echo_level = BaseType::GetEchoLevel();
             // Iterative solution for pressure
-            for(unsigned int it = 0; it < mMaxPressureIter; ++it)
-            {
-                if ( BaseType::GetEchoLevel() > 1 && rModelPart.GetCommunicator().MyPID() == 0)
-                    std::cout << "Pressure iteration " << it << std::endl;
-
-                NormDp = this->SolveStep();
-
-                Converged = this->CheckPressureConvergence(NormDp);
-
-                if ( Converged )
-                {
-                    if ( BaseType::GetEchoLevel() > 0 && rModelPart.GetCommunicator().MyPID() == 0)
-                        std::cout << "Predictor-corrector converged in " << it+1 << " iterations." << std::endl;
+            for (unsigned int it = 0; it < mMaxPressureIter; ++it) {
+                KRATOS_INFO_IF("FSStrategy", echo_level > 1) << "Pressure iteration " << it << std::endl;
+                norm_dp = this->SolveStep();
+                converged = this->CheckPressureConvergence(norm_dp);
+                if (converged) {
+                    KRATOS_INFO_IF("FSStrategy", echo_level > 0) << "Predictor-corrector converged in " << it + 1 << " iterations." << std::endl;
                     break;
                 }
             }
-            if (!Converged && BaseType::GetEchoLevel() > 0 && rModelPart.GetCommunicator().MyPID() == 0)
-                std::cout << "Predictor-correctior iterations did not converge." << std::endl;
-
-        }
-        else
-        {
+            KRATOS_WARNING_IF("FSStrategy", !converged && echo_level > 0) << "Predictor-correctior iterations did not converge." << std::endl;
+        } else {
             // Solve for fractional step velocity, then update pressure once
-            NormDp = this->SolveStep();
+            norm_dp = this->SolveStep();
+            // If not doing predictor corrector iterations, norm_dp will
+            // typically be "large" since we are not iterating on pressure.
+            // It makes no sense to report that the iteration didn't converge
+            // based on this.
+            converged = true;
```

Comment:
but the velocity can still not be converged right?
So unless I am using predictor corrector it will always tell that it converged?

Not sure if this is a good idea

But maybe I misunderstood

Reply:
The velocity convergence is check in the `SolveStep`, which throws a warning in case the momentum splitting does not convergence.

This was already there. I think that the original point was that you likely do not converge in terms of the pressure. As this is "intentionally" done for the sake of efficiency, checking the pressure norm and throwing a warning in the majority of the cases makes no sense at all. We can of course change it, but I'd rather keep it as it is.

Reply:
We can return the convergence state from `SolveStep` . that is the convergence of velocity iterations.

Reply:
> We can return the convergence state from `SolveStep` . that is the convergence of velocity iterations.

that sounds like the logical thing to do for me

Reply:
OK. We can change the return type of the `SolveStep` to do this.

Reply:
Done. Thanks guys!